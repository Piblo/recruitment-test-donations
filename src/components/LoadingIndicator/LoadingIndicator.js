import styled, { keyframes } from 'styled-components';

const leftDot = keyframes`
	0% {background-color: rgba(255, 255, 255, .2); }
	25% { background-color: rgba(255, 255, 255, 1); }
	50% { background-color: rgba(255, 255, 255, .2); }
	75% { background-color: rgba(255, 255, 255, .2); }
	100% { background-color: rgba(255, 255, 255, .2); }
`;

const middleDot = keyframes`
	0%{background-color: rgba(255, 255, 255, .2);}
	25%{background-color: rgba(255, 255, 255, .2);}
	50%{background-color: rgba(255, 255, 255, 1);}
	75%{background-color: rgba(255, 255, 255, .2);}
	100%{background-color: rgba(255, 255, 255, .2);}
`;

const rightDot = keyframes`
	0%{background-color: rgba(255, 255, 255, .2);}
	25%{background-color: rgba(255, 255, 255, .2);}
	50%{background-color: rgba(255, 255, 255, .2);}
	75%{background-color: rgba(255, 255, 255, 1);}
	100%{background-color: rgba(255, 255, 255, .2);}
`;

const LoadingIndicator = styled.div`
	position: absolute;
	width: 12px;
	height: 12px;
	top: 50%;
	left: 50%;
	border-radius: 12px;
	animation: ${middleDot} 3s ease-in-out infinite;

  &:before {
    content: "";
    position: absolute;
    top: 0px;
    left: -25px;
    height: 12px;
    width: 12px;
    border-radius: 12px;
    animation: ${leftDot} 3s ease-in-out infinite;
  }

  &:after {
    content: "";
    position: absolute;
    top: 0px;
    left: 25px;
    height: 10px;
    width: 10px;
    border-radius: 10px;
    animation: ${rightDot} 3s ease-in-out infinite;
  }
`;

export default LoadingIndicator;
