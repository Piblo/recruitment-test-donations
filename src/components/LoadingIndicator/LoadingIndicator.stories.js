import React from 'react';
import { storiesOf } from '@storybook/react';
import LoadingIndicator from './LoadingIndicator';

storiesOf('LoadingIndicator', module)
  .add('default', () => (
    <LoadingIndicator />
  ));
