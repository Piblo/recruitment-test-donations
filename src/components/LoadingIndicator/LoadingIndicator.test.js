import React from 'react';
import renderer from 'react-test-renderer';
import LoadingIndicator from './LoadingIndicator';

describe('<LoadingIndicator>', () => {
  it('Matches snapshot', () => {
    const tree = renderer.create(<LoadingIndicator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
