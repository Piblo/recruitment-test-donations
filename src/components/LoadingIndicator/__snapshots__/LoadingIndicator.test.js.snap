// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingIndicator> Matches snapshot 1`] = `
<div
  className="LoadingIndicator-sc-1cg5c9w-0 JjQyC"
/>
`;
